### DISCUSSION TITLE
Unresolved bind-time variable !(bind.TimeStamp)

### DISCUSSION THREAD

**Discussion by Nobru25st** [wixtoolset/issues#8737] [Questions]
Hello,
I'm new to WIX, I got a project with Wix V3 toolset and I wanted to be able to retrieve the system date at runtime to put it in a variable. But I'm facing this bind-time not solving error. " Unresolved bind-time variable !(bind.TimeStamp)"

<Property Id="DATE" Value="!(bind.TimeStamp)" />

Can you help me?

**Comment by bevanweiss** [comment]
> !(bind.TimeStamp)"

Where did you find a reference to such a bind variable?
I would not expect there to be a reason for the Windows Installer to advertise the system date.  You probably want to read up on the Windows Installer concepts https://learn.microsoft.com/en-us/windows/win32/msi/windows-installer-portal
WiX is something of a 'compiler' around the Windows installer subsystem.

If you need something 'weird' like system date for an installation, then:

1. You need to really question whether it's appropriate.  Installers are supposed to be re-producible.  Introducing a transient element like system date seems like it would be a terrible idea.
2. You will likely need to implement it yourself as a [Custom Action.](https://stackoverflow.com/questions/20683741/how-to-add-custom-action-to-wix-setup-project)

**Reply by Nobru25st** [reply]
Thank you for your answer, in fact I thought I could retrieve and store the date of installation of the application more easily, to then be able to use it as part of an authorization to use the application.

**Reply by Nobru25st** [reply]
I found the reference to this variable by trusting AI-generated content too quickly! it will teach me ...
